Server/Components/ObjectOscillator.cpp

```diff
@@ -0,0 +1,66 @@
+#include "ObjectOscillator.h"
+#define TAU 6.283185
+
+ObjectOscillator::ObjectOscillator(GameObject * go, float period, float distance, Vector3 direction, float cooldown, float waitDelay) {
```

Comment:
the parametized names can be chanegd to pPeriod, pDistance, pDirection to avoid using this-> when setting member variables. I dont know if thats an extra call, but could do

Reply:
I think ill keep them as is, the performance will not be any different and keeping them makes their purpose *slightly* clearer